src/hooks/useENSRegistration.ts

```diff
@@ -69,21 +122,55 @@ export default function useENSRegistration({
   duration: number;
   name: string;
 }) {
-  const { nativeCurrency } = useAccountSettings();
+  const { nativeCurrency, accountAddress } = useAccountSettings();
   const isValidLength = useMemo(() => name.length > 2, [name.length]);
 
   const getRegistrationValues = useCallback(async () => {
     const isAvailable = await getAvailable(name);
     if (isAvailable) {
-      // we need the price only if is available
       const rentPrice = await getRentPrice(name, duration * secsInYear);
+      const nativeAssetPrice = ethereumUtils.getPriceOfNativeAssetForNetwork(
+        Network.mainnet
+      );
       const formattedRentPrice = formatRentPrice(
         rentPrice,
         duration,
-        nativeCurrency
+        nativeCurrency,
+        nativeAssetPrice
       );
+      const gasLimit = await estimateENSRegistrationGasLimit(
+        name,
+        accountAddress,
+        duration * secsInYear,
+        rentPrice.toString()
+      );
+
+      const {
+        gasFeeParamsBySpeed,
+        currentBaseFee,
+      } = await getEIP1559GasParams();
+
+      const formattedEstimatedNetworkFee = formatEstimatedNetworkFee(
+        gasLimit,
+        currentBaseFee.gwei,
+        gasFeeParamsBySpeed.normal.maxPriorityFeePerGas.gwei,
+        nativeCurrency,
+        nativeAssetPrice
+      );
+
+      const weiEstimatedTotalCost = add(
+        formattedEstimatedNetworkFee.wei,
+        formattedRentPrice.wei
+      );
+      const totalRegistrationCost = formatTotalRegistrationCost(
+        weiEstimatedTotalCost,
+        nativeCurrency,
+        nativeAssetPrice
+      );
```

Comment:
I reckon we should put this in a separate hook `useENSRegistrationGasEstimate`. The reason why is because `estimateENSRegistrationGasLimit` may take a while for feedback. I think it would be a better UX if the user can see the availability of their ENS while the gas estimate is loading in the background?

Reply:
🤝

Reply:
hm even tho i agree with you,  the estimation fee depends on the rent price which is showed together with the availability of the ens